tests/sentry/receivers/test_sentry_apps.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import absolute_import
+
+from mock import patch
+
+from sentry.testutils import APITestCase, TestCase
+from sentry.testutils.helpers.faux import faux
+
+
+@patch('sentry.tasks.sentry_apps.process_resource_change.delay')
+class TestIssueSaved(TestCase):
+    def test_processes_created_issues(self, delay):
+        issue = self.create_group()
+        assert faux(delay).called_with(
+            action='created',
+            sender='Group',
+            instance_id=issue.id,
+        )
+
+    def test_does_not_process_unless_created(self, delay):
+        issue = self.create_group()
+        delay.reset_mock()
+        issue.update(message='Stuff blew up')
+        assert len(delay.mock_calls) == 0
+
+
+# This testcase needs to be an APITestCase because all of the logic to resolve
+# Issues and kick off side effects are just chillin in the endpoint code -_-
```

Comment:
What about the code that lives in `Release.set_commits`? It resolves groups when release commits are processed.

Reply:
Just thought about this when Meredith and I were looking over it. Didn't know where that happened from, so thanks 😄 Also another reason to encapsulate domain logic somewhere central. Just saying :P

Reply:
I :100: agree on having places for logic that isn't buried in an endpoint.